consul/config_templates/consul-values-1.9.1-demo.yaml

```diff
@@ -0,0 +1,1841 @@
+  priorityClassName: ""
+
+# Mesh Gateways enable Consul Connect to work across Consul datacenters.
+meshGateway:
+  # If mesh gateways are enabled, a Deployment will be created that runs
+  # gateways and Consul Connect will be configured to use gateways.
+  # See https://www.consul.io/docs/connect/mesh_gateway.html
+  # Requirements: consul 1.6.0+ and consul-k8s 0.15.0+ if using
+  # global.acls.manageSystemACLs.
+  enabled: false
+
+  # Globally configure which mode the gateway should run in.
+  # Can be set to either "remote", "local", "none" or empty string or null.
+  # See https://consul.io/docs/connect/mesh_gateway.html#modes-of-operation for
+  # a description of each mode.
+  # If set to anything other than "" or null, connectInject.centralConfig.enabled
+  # should be set to true so that the global config will actually be used.
+  # If set to the empty string, no global default will be set and the gateway mode
+  # will need to be set individually for each service.
+  globalMode: local
+
+  # Number of replicas for the Deployment.
+  replicas: 2
+
+  # What gets registered as WAN address for the gateway.
+  wanAddress:
+    # source configures where to retrieve the WAN address (and possibly port)
+    # for the mesh gateway from.
+    # Can be set to either: `Service`, `NodeIP`, `NodeName` or `Static`.
+    #
+    # - `Service` - Determine the address based on the service type.
+    #
+    #   - If `service.type=LoadBalancer` use the external IP or hostname of
+    #     the service. Use the port set by `service.port`.
+    #
+    #   - If `service.type=NodePort` use the Node IP. The port will be set to
+    #     `service.nodePort` so `service.nodePort` cannot be null.
+    #
+    #   - If `service.type=ClusterIP` use the `ClusterIP`. The port will be set to
+    #     `service.port`.
+    #
+    #   - `service.type=ExternalName` is not supported.
+    #
+    # - `NodeIP` - The node IP as provided by the Kubernetes downward API.
+    #
+    # - `NodeName` - The name of the node as provided by the Kubernetes downward
+    #   API. This is useful if the node names are DNS entries that
+    #   are routable from other datacenters.
+    #
+    # - `Static` - Use the address hardcoded in `meshGateway.wanAddress.static`.
+    source: "Service"
+
+    # Port that gets registered for WAN traffic.
+    # If source is set to "Service" then this setting will have no effect.
+    # See the documentation for source as to which port will be used in that
+    # case.
+    port: 443
+
+    # If source is set to "Static" then this value will be used as the WAN
+    # address of the mesh gateways. This is useful if you've configured a
+    # DNS entry to point to your mesh gateways.
+    static: ""
+
+  # The service option configures the Service that fronts the Gateway Deployment.
+  service:
+    # Whether to create a Service or not.
+    enabled: true
+
+    # Type of service, ex. LoadBalancer, ClusterIP.
+    type: LoadBalancer
```

Comment:
If we did not have a public cluster environment of the public cloud provider, the type of service may not work well. Maybe the ClusterIP can work and NodePort to expose the port

Reply:
Yes, but it is not enabled in this configuration. It is for Mesh Gateway, used to federate Consul datacenters.

Reply:
Does it was not enabled by line 1487?

Reply:
No, Mesh Gateway is enabled on line 1002 (`meshgateway.enabled`).

Reply:
Thanks